examples/benchmark.py

```diff
@@ -127,7 +127,6 @@
 
         for params in params_sweep:
             _, _, _, res = run(**params)
-            clear_cache()
```

Comment:
1. If I am not mistaken, this was required because the `SymbolicCache` implemented in our code was causing a memory leak.
2. With the new implementation, calling this calls `__del__` on our objects which frees the memory associated with them. In the next iteration when you try to access last_res, the object doesn't exist because we specifically `free`ed the associated memory and then called the garbage collector inside this `clear_cache` method. 
3. The implementation of memory management in this PR doesn't maintain a reference count, i.e. tracking the number of references to a particular object since I believe python does that very well at its end and calls `__del__` automatically when there are no remaining references. I believe the new implementation is actually anxious in freeing memory so explicit calls to clear the cache should no longer be required, especially since the deep copy has also been removed from the caching mechanism. As far as I know, this call was always a work-around for a bug in our caching implementation (causing the memory leak) since in the normal scenario we should be able to rely on python's GC to know when to clear memory. I thought I shouldn't have to work around a work-around for a bug that has been fixed - that's why I removed it. 

How else would you rather proceed?